go/vt/tabletserver/schema_info.go

```diff
@@ -186,23 +186,37 @@ func (si *SchemaInfo) Open(appParams, dbaParams *sqldb.ConnParams, schemaOverrid
 
 	tables := make(map[string]*TableInfo, len(tableData.Rows)+1)
 	tables["dual"] = &TableInfo{Table: schema.NewTable("dual")}
+	wg := sync.WaitGroup{}
+	mu := sync.Mutex{}
 	for _, row := range tableData.Rows {
-		tableName := row[0].String()
-		tableInfo, err := NewTableInfo(
-			conn,
-			tableName,
-			row[1].String(), // table_type
-			row[3].String(), // table_comment
-			si.cachePool,
-		)
-		if err != nil {
-			si.recordSchemaError(err, tableName)
-			// Skip over the table that had an error and move on to the next one
-			continue
-		}
-		tableInfo.SetMysqlStats(row[4], row[5], row[6], row[7])
-		tables[tableName] = tableInfo
+		wg.Add(1)
+		go func(row []sqltypes.Value) {
+			defer wg.Done()
+
+			conn := getOrPanic(ctx, si.connPool)
```

Comment:
The other option is to use a separate set of connections here. Otherwise, we'd be wasting connpool connections here.